### PR TITLE
Update Snaps documentation

### LINKED ISSUE #19: Update Snaps API documentation
Clean up the Snaps API and exports reference documentation and make sure all the methods are documented accurately, consistently, and clearly. Format should ideally be consistent with the other API docs (see #13 and #12).

### PR DESCRIPTION
**Pull request by alexandratran**
Update Snaps documentation to integrate recent changes made in the legacy Snaps docs. Edit these changes to match the new documentation standards and structure, and make edits for clarity, consistency, and style throughout the Snaps docs.

Changes from the following PRs were integrated and further edited:
- https://github.com/MetaMask/metamask-docs/pull/588
- https://github.com/MetaMask/metamask-docs/pull/590
- https://github.com/MetaMask/metamask-docs/pull/595
- https://github.com/MetaMask/metamask-docs/pull/603
- https://github.com/MetaMask/metamask-docs/pull/607
- https://github.com/MetaMask/metamask-docs/pull/616
- https://github.com/MetaMask/metamask-docs/pull/623
- https://github.com/MetaMask/metamask-docs/pull/653
- https://github.com/MetaMask/metamask-docs/pull/656
- https://github.com/MetaMask/metamask-docs/pull/661
- https://github.com/MetaMask/metamask-docs/pull/663
- https://github.com/MetaMask/metamask-docs/pull/671
- https://github.com/MetaMask/metamask-docs/pull/638
- https://github.com/MetaMask/metamask-docs/pull/644
- https://github.com/MetaMask/metamask-docs/pull/589
- https://github.com/MetaMask/metamask-docs/pull/605

Fixes #19

### CODE REVIEW

**Comment by alexandratran** [comment]
@metamaskbot publish-preview

**Comment by alexandratran** [comment]
@metamaskbot publish-preview

File: snaps/get-started/quickstart.md [hunk 1]
```diff
@@ -61,13 +65,13 @@ You can customize your snap by editing `index.ts` in the `packages/snap/src` fol
 [`snap_confirm`](../reference/rpc-api.md#snap_confirm) method to display a custom confirmation screen:
```
Comment (ziad-saab) [inline snaps/get-started/quickstart.md:65]: I know this isn't part of your PR, but for some reason I'm seeing a bit of outdated docs. As `snap_confirm` has been deprecated, the template snap now uses the `snap_dialog` method to do its work:

https://github.com/MetaMask/template-snap-monorepo/blob/main/packages/snap/src/index.ts

File: snaps/how-to/request-permissions.md [hunk 2]
```diff
@@ -10,108 +14,26 @@ and you must specify a snap's required permissions in the `initialPermissions` f
...
 
 ```json
-...
 "initialPermissions": {
   "endowment:long-running": {}
 },
-...
-```
-
-### `endowment:network-access`
```
Comment (ziad-saab) [inline snaps/how-to/request-permissions.md:41]: Why have all these been removed?

Comment (ziad-saab) [reply snaps/how-to/request-permissions.md:41]: Resolved. I see where you added them.

**Review by ziad-saab — APPROVED** [review]

File: snaps/concepts/anatomy.md [hunk 3]
```diff
@@ -82,7 +87,8 @@ await ethereum.request({
   ],
 });
 
-const hello = await ethereum.request({
+// Invoke the "hello" RPC method exposed by the snap
+const hello = await window.ethereum.request({
   method: 'wallet_invokeSnap',
```
[the remark this answers is not in the document]
Comment (alexandratran) [reply snaps/concepts/anatomy.md:92]: Thanks! I just missed this change in the legacy docs.

File: snaps/get-started/quickstart.md (hunk 1 above)
Comment (alexandratran) [reply snaps/get-started/quickstart.md:65]: Thanks! This change wasn't made in the legacy docs, but I've added it to these new docs now.

File: snaps/reference/permissions.md [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+            }
+          }
+        }
+      ]
+    }
+  }
+}
+```
+
+## endowment:ethereum-provider
```
Comment (bgravenorst) [inline snaps/reference/permissions.md:53]: Should this be formatted?

File: snaps/how-to/use-custom-ui.md [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+]);
+
+return content;
+```
+
+## Components
+
+The `NodeType` enum exported by `@metamask/snaps-ui` details the available components.
+
+### copyable
```
[the remark this answers is not in the document]
Comment (alexandratran) [reply snaps/how-to/use-custom-ui.md:38]: For some reason, in the MM docs, all formatted headers render at the same small size (same as formatted body text), so formatted headers look funny:
Screenshot 2023-03-21 at 11 18 44 AM
So for now, I opted to not format any headers to keep the variable sizing. This should be fixable, but from a cursory look at the Docusaurus docs and CSS files, I can't figure it out. I'll wait till Josh comes back to ask him, or ask one of the engineers during the MM doc-a-thon.

**Pull request merged**